### PR TITLE
Support long string switch

### PR DESCRIPTION
**Pull request by lightszero**
fixed https://github.com/neo-project/neo-devpack-dotnet/issues/86

add long string switch support.

find IL code pattern for switch,and delete hashpart,it work for current unittest code.

```c#
        public static object Main(string method, object[] args)
        {
            switch (method)
            {
                case "0": return 1;
                case "1": return 2;
                case "2": return 3;
                case "3": return 4;
                case "4": return 5;
                case "5": return 6;
                case "6": return 7;
                case "7": return 8;
                case "8": return 9;
                case "9": return 10;
                case "10": return 11;
                case "11": return 12;
                case "12": return 13;
                case "13": return 14;
                case "14": return 15;
                case "15": return 16;
                case "16": return 17;
                case "17": return 18;
                case "18": return 19;
                case "19": return 20;
                case "20": return 21;

                default: return 99;
            }
        }
```

### CODE REVIEW

**Comment by erikzhang** [comment]
I think we should create `ComputeStringHash()` in NeoContract Interop Services first.

File: src/Neo.Compiler.MSIL/MSIL/Conv_Multi.cs [hunk 1]
```diff
@@ -964,6 +957,132 @@ private int _ConvertCall(OpCode src, NeoMethod to)
...
+                        isjumptable = true;
+                        jumptableaddr = method.GetNextCodeAddr(code3.addr);
+                        skipcount += 3;
+                        jumptablecount++;
+                    }
+                    else
+                    {
+                        isjumptable = false;
+                        //順便看看是不是ldstr 段
+                        if (bLdLoc1 && code1.tokenI32 == last.tokenI32 && code2.code == CodeEx.Ldstr && (code3.code == CodeEx.Call || code3.code == CodeEx.Callvirt) && code3.tokenMethod.Contains("String::op_Equality"))
```
Comment (shargon) [inline src/Neo.Compiler.MSIL/MSIL/Conv_Multi.cs]: could be replaced `Contains` by a regex? @lightszero

**Comment by lightszero** [comment]
> I think we should create `ComputeStringHash()` in NeoContract Interop Services first.

https://github.com/neo-project/neo-vm/pull/15/files
long long ago

File: src/Neo.Compiler.MSIL/MSIL/Conv_Multi.cs (hunk 1 above)
Comment (lightszero) [reply src/Neo.Compiler.MSIL/MSIL/Conv_Multi.cs]: i am worse about regex

**Comment by erikzhang** [comment]
@lightszero I mean create it in NeoContract Interop Services, not in NeoVM.

**Comment by shargon** [comment]
Maybe create it in the compiler only if is needed is better, because maybe in different frameworks dotnet3, dotnet2, it could change the behavior.

**Comment by lightszero** [comment]
but this pr is skip opcode for " ComputeStringHash" 
 I think for switch we do not need to implentment ComputeStringHash
@erikzhang

**Comment by shargon** [comment]
I agree, if we can detect and remove the `ComputeStringHash` we can do it, because every time after this check they check again the right value

**Comment by erikzhang** [comment]
That's great. Please resolve the conflicts.

File: src/Neo.Compiler.MSIL/MSIL/Conv_Multi.cs (hunk 1 above)
Comment (shargon) [reply src/Neo.Compiler.MSIL/MSIL/Conv_Multi.cs]: I will do it

Comment (shargon) [reply src/Neo.Compiler.MSIL/MSIL/Conv_Multi.cs]: https://github.com/neo-project/neo-devpack-dotnet/pull/138 @lightszero

Comment (lightszero) [reply src/Neo.Compiler.MSIL/MSIL/Conv_Multi.cs]: tks a lot,  megre this first .
I am learning regex too.

**Comment by lightszero** [comment]
why travis always error?
@neo-project/core 
help

File: src/Neo.Compiler.MSIL/MSIL/Conv_Multi.cs (hunk 1 above)
Comment (shargon) [reply src/Neo.Compiler.MSIL/MSIL/Conv_Multi.cs]: It's a PR to your branch

**Comment by shargon** [comment]
@lightszero it was because it's required to execute `dotnet format` first in order to follow all of us the same standards.

**Comment by shargon** [comment]
I think that it require a `rebase`, a lot of files looks as changed, when are not

**Comment by lightszero** [comment]
i had do some change on unittest, for test contract on  debug and release.so got many file changes

**Comment by shargon** [comment]
I don't know why appear files as changed, when are not changed :S

**Comment by lock9** [comment]
@lightszero  @shargon  please check #141 . I rebased this PR and removed some problematic merges.

**Pull request closed without merging**